src/pages/page4/page4.ts

```diff
@@ -35,65 +33,25 @@ export class Page4 {
 	    this.notification_toggle = value;
     });
 
+    this.storage.get('sound_toggle').then((value) => {
+      this.sound_toggle = value;
```

Comment:
So instead of grabbing all these variables from memory, you create one variable (object) to hold the values and just grab that. I also recommend that you use the user provider which handles all of this.